google_photos_takeout_helper/__main__.py

```diff
@@ -233,34 +261,26 @@ def find_duplicates(path: Path, filter_fun=lambda file: True):
 
                 files_by_full_hash[full_hash].append(file)
 
-        # Now we have the final multimap of absolute dups, We now can attempt to find the original file
+    # Removes all duplicates in folder
+    def remove_duplicates(dir: Path):
+        find_duplicates(dir, lambda f: (is_photo(f) or is_video(f)))
+        nonlocal s_removed_duplicates_count
+
+        # Now we have populated the final multimap of absolute dups, We now can attempt to find the original file
         # and remove all the other duplicates
         for files in files_by_full_hash.values():
             if len(files) < 2:
                 continue # this file size is unique, no need to spend cpu cycles on it
-            original = None
-            for file in files:
-                if not _re.search(r'\(\d+\).', file.name):
-                    original = file
-            if original is None:
-                original = files[0]
 
-            dups = files.copy()
-            dups.remove(original)
-            duplicates += dups
-
-        return duplicates
-
-    # Removes all duplicates in folder
-    def remove_duplicates(dir: Path):
-        duplicates = find_duplicates(dir, lambda f: (is_photo(f) or is_video(f)))
-        for file in duplicates:
-            file.unlink()
-        nonlocal s_removed_duplicates_count
-        s_removed_duplicates_count += len(duplicates)
+            s_removed_duplicates_count += len(files) - 1
+            for file in files:
+            #TODO reconsider which dup we delete these now that we're searching globally?
+                if len(files) > 1:
+                    file.unlink()
+                    files.remove(file)
         return True
 
-    # PART 2: Fixing metadata and date-related stuff
+    # PART 1: Fixing metadata and date-related stuff
 
     # Returns json dict
     def find_json_for_file(file: Path):
```

Comment:
A simple way to handle the shit google json names is to replace this find_json_for_file function with this.
```
# Returns json dict
    def find_json_for_file(file: Path):
        potential_json = list(file.parent.rglob(file.stem + ".*json"))
        if len(potential_json) != 0:
            try:
                with open(potential_json[0], 'r') as f:
                    dict = _json.load(f)
                return dict
            except:
                raise FileNotFoundError(f"Couldn't find json for file: {file}")
        else:
            raise FileNotFoundError(f"Couldn't find json for file: {file}")
```

It just enumerates any possible extension they may have jammed in there. In practice I've always seen this to be just the single json file but theorityically this could enumerate multiple if you had two files like IMG_1.jpg and IMG_1.png in the same origin directory. This seems rare enough.

Reply:
This should be for another PR IMO

Reply:
Yes, I will implement this in another PR just after we merge this